plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java

```diff
@@ -233,24 +239,51 @@ public GeoIpProcessor doCreate(String processorTag, Map<String, Object> config)
             String databaseFile = readStringProperty(TYPE, processorTag, config, "database_file", "GeoLite2-City.mmdb");
             List<String> propertyNames = readOptionalList(TYPE, processorTag, config, "properties");
 
-            final Set<Property> properties;
+            DatabaseReader databaseReader = databaseReaders.get(databaseFile);
+            if (databaseReader == null) {
+                throw newConfigurationException(TYPE, processorTag, "database_file", "database file [" + databaseFile + "] doesn't exist");
+            }
+
+            String databaseType = databaseReader.getMetadata().getDatabaseType();
+
+            // TODO(talevy): make this whole thing more DRY
+            Set<Property> properties;
             if (propertyNames != null) {
                 properties = EnumSet.noneOf(Property.class);
                 for (String fieldName : propertyNames) {
                     try {
-                        properties.add(Property.parse(fieldName));
-                    } catch (Exception e) {
+                        final Property property;
+                        // TODO(talevy): figure out way to switch type without having to worry about valid type input (use if statements)
+                        switch (databaseType) {
+                            case CITY_DB_TYPE:
+                                property = Property.parseCityProperty(fieldName);
+                                break;
+                            case COUNTRY_DB_TYPE:
+                                property = Property.parseCountryProperty(fieldName);
+                                break;
+                            default:
+                                throw newConfigurationException(TYPE, processorTag, "database_file", "Unsupported database type [" + databaseType + "]");
+                        }
+                        properties.add(property);
+                    } catch (IllegalArgumentException e) {
+                        // TODO(talevy): add databaseType context to this exception message
                         throw newConfigurationException(TYPE, processorTag, "properties", "illegal field option [" + fieldName + "]. valid values are [" + Arrays.toString(Property.values()) + "]");
                     }
                 }
             } else {
-                properties = DEFAULT_PROPERTIES;
+                // TODO(talevy): figure out way to switch type without having to worry about valid type input (use if statements)
```

Comment:
@martijnvg, you have no problems with this unnecessary (but safe) usage of the switch along with the exception throwing?